### ISSUE TITLE
Accuracies saved in history.history after model.fit() are different than the ones printed on-screen during training

### ISSUE THREAD

**Issue by malessandrini** [tensorflow/tensorflow#48408]
**System information**
Using Colab with Keras 2.4.0 and TensorFlow 2.4.1

**Describe the current behavior**
After calling "history = model.fit(...)", values saved in history.history['loss'] and history.history['accuracy'] are different from the ones printed on-screen at every epoch during training.

**Describe the expected behavior**
Values should be the same, as is with previous tf versions and with other parameters (like val_accuracy)

**Standalone code to reproduce the issue**
Shared link: https://colab.research.google.com/drive/14Uogeq8wRlZlinaKLbkFr_Bl2aLzUJuy?usp=sharing
(slightly modified version of MNIST tutorial, see the block right after calling model.fit())

See also original discussion: https://stackoverflow.com/questions/67001654/accuracy-in-history-dictionary-different-from-what-printed-on-screen

**Comment by jvishnuvardhan** [comment]
@malessandrini This is a known regression issue and is tracked with this issue https://github.com/tensorflow/tensorflow/issues/48033

Duplicate of #https://github.com/tensorflow/tensorflow/issues/48033

**Comment by jvishnuvardhan** [comment]
@malessandrini This was resolved in recent tf-nightly. I couldn't reproduce the error anymore. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/ea9c97436cf9949b21df32a17eb42969/copia-di-beginner.ipynb).

`loss` and accuracy from `model.fit`

```
Epoch 1/5
1875/1875 [==============================] - 6s 3ms/step - loss: 0.3012 - accuracy: 0.9124
Epoch 2/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.1457 - accuracy: 0.9561
Epoch 3/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.1091 - accuracy: 0.9670
Epoch 4/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0888 - accuracy: 0.9726
Epoch 5/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0768 - accuracy: 0.9756
```

`loss` and `accuracy` from `history` object
```
`loss`          :  [0.3012, 0.1457, 0.1091, 0.0888, 0.0768]
`accuracy` :  [0.9124, 0.9561, 0.967, 0.9726, 0.9756]
```

Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by aryamohan23** [comment]
The same issue is happening again in tf2.8. 
I have a custom training loop with the `train_step()` function returning 2 losses: d_loss and g_loss.
The values turn out to be different when I call `model.fit()` - 

```
Epoch 1/2
110/110 [==============================] - 33s 294ms/step - d_loss: **-1.5941** - g_loss: **2.2890**
Epoch 2/2
110/110 [==============================] - 32s 288ms/step - d_loss: **-2.5873** - g_loss: **2.8013**
```
```
print([round(x, 4) for x in history_init.history['d_loss']])
print([round(x, 4) for x in history_init.history['g_loss']])
```
OUTPUT:
d_loss: [-2.7844, -2.2764]
g_loss: [2.4266, 3.3268]

**Comment by VladimirVisnovsky** [comment]
Error still persists in 2.12.0.

**Comment by tombinic** [comment]
Error present also in tf 2.16.1
